### ISSUE TITLE
Plans for reference-types proposal?

### ISSUE THREAD

**Issue by granstrom** [fastly/lucet#265]
Is there any interest in adding support for [reference-types](https://github.com/WebAssembly/reference-types/blob/master/proposals/reference-types/Overview.md) to Lucet? If so, who should I speak with?

Thanks!

**Comment by acfoltzer** [comment]
Yes, we're definitely interested in supporting these and related proposals such as exceptions. Right now, we're following the progress of https://github.com/CraneStation/cranelift/pull/871, which adds reference types to the Cranelift compiler backend that we use in `lucetc`. Once that is merged, we'll be able to move forward on supporting them in Lucet.

If you're interested in helping out, we'd be happy to collaborate on here, or via some of the Discord communities around Rust and WebAssembly. I'm `acfoltzer#6411` on there.

**Comment by granstrom** [comment]
Thanks Adam! We may be interested in collaborating on this: I'll sync with the team and get back to you (over some channel).

**Comment by EliSnow** [comment]
Looks like the upstream reference-types PR has landed

**Comment by pchickey** [comment]
As part of forming the Bytecode Alliance, we've announced that Lucet will be merging into the [Wasmtime codebase](https://github.com/bytecodealliance/wasmtime) - all of the unique capabilities of Lucet (AOT compilation, pre-allocated instance memory pools, separate execution stack with suspend/resume, and probably some other minor stuff) will be ported into Wasmtime. We don't have a timeline for it yet because we're still laying groundwork, but, we don't expect to develop any major new features like reference types as part of Lucet.

If you need some of the unique features of Lucet and want to join the effort to bring those into Wasmtime, lets please coordinate our efforts!

**Comment by EliSnow** [comment]
Very cool. One thing I found particularly awesome in Lucet is the yield capabilities. I hope those make it into Wasmtime!
